docs/sql/statements/copy-from.rst

```diff
@@ -114,6 +116,10 @@ URI
 A string literal or array of string literals containing URIs. Each URI must be
 formatted according to the `URI Scheme`_.
 
+The ``COPY FROM`` statement allows ``URIs`` with wildcards to conveniently match all
+directories and files. So, ``file:///tmp/import_data/*/*.json`` would expand to include
+every files with JSON extension within the immediate sub-directories of ``/tmp/import_data``.
+
```

Comment:
```suggestion
You can use filename `globbing`_ (i.e., ``*`` wildcards) with the ``COPY
FROM`` statement to construct URIs that can match multiple directories and
files.

Suppose you used ``file:///tmp/import_data/*/*.json`` as the URI. This URI
would match all JSON files located in subdirectories of the
``/tmp/import_data`` directory.

So, for example, these files would match:

- ``/tmp/import_data/foo/1.json``
- ``/tmp/import_data/bar/2.json``
- ``/tmp/import_data/1/boz.json``

.. CAUTION::

    A file named ``/tmp/import_data/foo/.json`` would also match the
    ``file:///tmp/import_data/*/*.json`` URI. The ``*`` wildcard matches any
    number of characters, including none.

However, these files would match:

- ``/tmp/import_data/1.json`` (two few subdirectories)
- ``/tmp/import_data/foo/bar/2.json`` (too many subdirectories)
- ``/tmp/import_data/1/boz.js`` (incorrect file extension)

.. _globbing: https://en.wikipedia.org/wiki/Glob_(programming)
```

how about this?

(I would move the `globbing` link definition to the bottom of the file

Reply:
Thanks @norosa, I have changed a few shown bolded below.
my understanding is that filename is /tmp/**a.json** and pathname is **/tmp/a.json**
Also wildcards include * and few other things, and I think we should clearly say only * is accepted.


You can use **`globbed`_ pathnames (specifically,** ``*`` wildcards) with the ``COPY
FROM`` statement to construct URIs that can match multiple directories and
files.
Suppose you used ``file:///tmp/import_data/*/*.json`` as the URI. This URI
would match all JSON files located in subdirectories of the
``/tmp/import_data`` directory.
So, for example, these files would match:

- ``/tmp/import_data/foo/1.json``
- ``/tmp/import_data/bar/2.json``
- ``/tmp/import_data/1/boz.json``
.. CAUTION::
    A file named ``/tmp/import_data/foo/.json`` would also match the
    ``file:///tmp/import_data/*/*.json`` URI. The ``*`` wildcard matches any
    number of characters, including none.
However, these files would not match:

- ``/tmp/import_data/1.json`` (two few subdirectories)
- ``/tmp/import_data/foo/bar/2.json`` (too many subdirectories)
- ``/tmp/import_data/1/boz.js`` (**file extension mismatch**)

Reply:
"i.e.," means the same as "specifically", so I would keep "i.e.," (you might've been thinking of "e.g.," which means "for example")

you are right about the difference between _filenames_ and _paths_. I would keep the grammar of the opening sentence though, so: "You can use pathname globbing...", because:

- "You can use globbed pathnames..." does not specify who is globbing (passive voice)
- "You can use pathname globbing..." makes it clear that the user can do the globbing (active voice)

"file extension mismatch" is a good change!

Reply:
I just noticed this was already merged. my bad! I am working on a small PR to add some copyedits to this file and will add you as a reviewer @jeeminso

Reply:
copyedits here: https://github.com/crate/crate/pull/11449